deepspeed/runtime/engine.py

```diff
@@ -1746,11 +1747,18 @@ def is_gradient_accumulation_boundary(self):
         """Query whether the current micro-batch is at the boundary of
         gradient accumulation, and thus will trigger gradient reductions and
         an optimizer step.
-
         Returns:
             bool: if the current step is a gradient accumulation boundary.
         """
-        return (self.micro_steps + 1) % self.gradient_accumulation_steps() == 0
+        if self._is_gradient_accumulation_boundary is None:
+            return (self.micro_steps + 1) % \
+                self.gradient_accumulation_steps() == 0
+        else:
+            return self._is_gradient_accumulation_boundary
+
+    def set_gradient_accumulation_boundary(self, is_boundary):
```

Comment:
I think a doc string is needed here to help clients use properly. In particular, about ensuring that clients clear this flag after optimizer.step().

Reply:
An alternative approach that comes to mind is instead of adding a new engine function, we can add a `is_gradient_accumulation_boundary` argument to `backward()` and `step()` such as: 
```python
   def backward(self, ..., is_gradient_accumulation_boundary=False):
         self._is_gradient_accumulation_boundary = is_gradient_accumulation_boundary
        ...

  def step(self, ..., is_gradient_accumulation_boundary=False):
        self._is_gradient_accumulation_boundary = is_gradient_accumulation_boundary
       ...
```
I am not sure if this will work well with the current kwargs constraints of `backward` and `step`.